MultiLabel/Core/src/main/java/org/tribuo/multilabel/ImmutableMultiLabelInfo.java

```diff
@@ -190,6 +190,23 @@ public int hashCode() {
         return new ImmutableInfoIterator(idLabelMap);
     }
 
+    @Override
+    public boolean domainAndIDEquals(ImmutableOutputInfo<MultiLabel> other) {
+        if (size() == other.size()) {
+            for (Map.Entry<Integer,String> e : idLabelMap.entrySet()) {
+                MultiLabel otherLbl = other.getOutput(e.getKey());
+                if (otherLbl == null) {
+                    return false;
+                } else if (!otherLbl.getLabelString().equals(e.getValue())) {
```

Comment:
I don't recall, does `MultiLabel` keep it's label string consistent?

Reply:
Yeah, because the domain is actually one `MultiLabel` per dimension and the `MutableMultiLabelInfo.observe` method looks at each label dimension individually and expands the domain that way.